### ISSUE TITLE
Fateful Encounter Issues in Pokemon Colosseum

### ISSUE THREAD

**Issue by ReclaimerShawn** [kwsch/PKHeX#2289]
So I was using PKHex to mod pokemon on Colosseum to transfer them over to my Gameboy and whenever I ticked the "Fateful Encounter" flag on Pokemon, the game would delete them upon it starting. I think for Deoxys to be legit and obey that the fateful encounter flag has to be checked, preventing a legit Deoxys from spawning in-game via PKHex. Taking off the Fateful Encounter flag fixes this issue partially, but considering Deoxys isn't legit without this flag, it isn't able to be transferred over to a Gameboy. This even occurs with the pokemon in the Encounters Database. A save file for reference will be attached. This is on the latest version of PKHex on a retail disc and a modded Wii by the way.
[Save File for Reference.zip](https://github.com/kwsch/PKHeX/files/3124678/Save.File.for.Reference.zip)

**Comment by kwsch** [comment]
Could be the game cleaning out data if the data isn't valid. Colo & XD structures weren't thoroughly mapped; maybe the "Fateful Encounter" flag is stored elsewhere.

https://github.com/kwsch/PKHeX/blob/b1ead9e63070731bd0bc9481d2866f8ab170269b/PKHeX.Core/PKM/XK3.cs#L62-L72

https://github.com/kwsch/PKHeX/blob/b1ead9e63070731bd0bc9481d2866f8ab170269b/PKHeX.Core/PKM/CK3.cs#L176-L178

Would have to do a before/after; transfer a fateful & nonfateful pkm and check the raw data?

**Comment by ReclaimerShawn** [comment]
I turned on the Fateful Encounter flag on a Latias I had caught via PKHex and then checked it out in PkmGCEditor and found that PKHex's FatefulEncounter option (at least for Colosseum) is actually the Invalid Pokemon flag. I have included both CK3 files and colopkm files (PKHex and PkmGCEditor respectively) to demonstrate this. These files can be found below:
[Latias Before and After.zip](https://github.com/kwsch/PKHeX/files/3125850/Latias.Before.and.After.zip)

PkmGCEditor can be found here: https://github.com/TuxSH/PkmGCTools/releases

Using PkmGCEditor, I turned off the Invalid Pokemon flag and turned on the obedient Pokemon flag and Deoxys appeared in the game. However, I still couldn't transfer it over to Pokemon Emerald. I encountered this issue with both Birth Island Deoxys and Space Center Deoxys. If I cannot transfer anything over to the gameboy, I'll try to use the Pomeg Glitch and get it that way.

**Comment by kwsch** [comment]
Any update on GBA->GC fateful/nonfateful transfer data comparisons?

**Comment by ReclaimerShawn** [comment]
Sorry - I ended up getting slammed with a week of College Finals. I'll get on to working on it now that I'm finished with everything.

**Comment by ijuintekka** [comment]
It seems the first bit of D7 and FB need to be set for fateful encounter in Colosseum.

**Comment by ReclaimerShawn** [comment]
These are the CK3 files of a Deoxys before and after the obedience bit is set:
[Deoxys Before and After.zip](https://github.com/kwsch/PKHeX/files/3160778/Deoxys.Before.and.After.zip)

The only change for me here was at FB.

**Comment by ijuintekka** [comment]
Using a code isn't the best way to test this as it isn't the way the game processes pokemon. I tested it by trading pokemon with and without the fateful encounter flag from Emerald and consistently the first bit of D7 and FB were set.

**Comment by ReclaimerShawn** [comment]
You're definitely right - this is just the only thing I've been able to use as I have no access to events - with the exception of Latias. The Latias in the file I uploaded previously had D7 set to 05 and FB set to 01.

**Comment by ijuintekka** [comment]
If D7 is 5 that means the first bit is already set, probably during trading. I'd need more information on the entirety of your process, from capture to trade, to comment on it further.

**Comment by ReclaimerShawn** [comment]
First, I used 3 iterations of the Pomeg Glitch and glitching my party slots to be able to scroll down to slot 255, allowing me to randomly set memory values to 1. I did this until the proper flags for the Southern Island event were set, which would allow me to access the event. I then went to the Island and battled it. I paralyzed it, and threw several Ultra Balls until it was caught. Then, after a while of having it, I sent it over to Pokemon Colosseum.

**Comment by ijuintekka** [comment]
If it's just a normal trade, that explains why D7 is already set. If my findings are correct, a non-fateful pokemon should never have the first bit set.

**Comment by kwsch** [comment]
Anything with 0xF8?
https://github.com/TuxSH/PkmGCTools/blob/2b98e7e86b00b7d6634fa21c8ce2819ac6e5e891/LibPkmGC/src/LibPkmGC/Colosseum/Common/Pokemon.cpp#L102-L107

**Comment by ijuintekka** [comment]
F8 has been zero in every case.

**Comment by ijuintekka** [comment]
Upon further testing with XD, I've determined that D7 being set was because of something pkhex was doing to GBA pokemon when fateful was checked and doesn't actually relate to fateful encounter.

FB remains the sole indicator in Colo. In Pokemon XD fateful encounter is set at the first bit of 33 and is present in all XD pokemon from the moment of capture. The byte is actually set to 3 before it is traded away for the first time and is 1 at all other times.

I've included a shot of XD's eevee at receipt and three times after being traded between games. Unaltered by pkhex at any point.

![eevee](https://user-images.githubusercontent.com/28776339/57568578-6290bf80-73e1-11e9-8395-352b438ecf56.png)
